### ISSUE TITLE
Copied text is lost after dialog is closed

### ISSUE THREAD

**Issue by notmatthancock** [enthought/traitsui#732]
To reproduce:

1. Run the code below
2. Copy the text to clipboard
3. Close the traits dialog
4. Try to paste
5. Be sad

```python
from traits.api import HasTraits, Str
from traitsui.api import Item, View


class Foo(HasTraits):
    x = Str('why hello there')
    traits_view = View(Item('x'))


if __name__ == '__main__':
    Foo().configure_traits()
```

**Comment by corranwebster** [comment]
I can't reproduce on OS X - what platform are you on?

**Comment by corranwebster** [comment]
This is most likely a Qt issue - when you do a copy from a text field that is entirely handled by Qt-level code (TraitsUI doesn't override field-level copy in any way).  Which version of Qt are you on?

**Comment by notmatthancock** [comment]
>  Which version of Qt are you on?
```bash
$ edm list | grep qt
pyqt_commercial        4.12.1-1       enthought/pyqt-commercial
qt                     4.8.7-11       enthought/free
```
But I can produce it with qt5 as well in a fresh environment.

**Comment by corranwebster** [comment]
Yes, I can see it on Qt5, but a fairly old version.  I think next steps in diagnosis:

- try more recent versions of PyQt
- try PySide2
- try to duplicate with pure PyQt/PySide

**Comment by corranwebster** [comment]
Quick test: I don't see this on TraitsUI running with a pip-installed PySide2 (you can check this by installing TraitsUI using `python etstool.py install --toolkit=pyside2` and running the example in the environment that is created by the command).

I think that's enough to say that this isn't a TraitsUI bug.

**Comment by notmatthancock** [comment]
> I think that's enough to say that this isn't a TraitsUI bug.

Fair enough. However, is there a natural place in the ETS stack where a fix (see, e.g., [here](https://stackoverflow.com/questions/2007103/how-can-i-disable-clear-of-clipboard-on-exit-of-pyqt-application) or [here](https://riverbankcomputing.com/pipermail/pyqt/2009-April/022857.html)) could be placed since this is a frequent annoyance? pyface maybe?

**Comment by corranwebster** [comment]
That's possible - but the code would need to be added in several places or we would need to set up some sort of generic listener for application shut-down.  Reasonable places might include:

- in the `close` method of `traitsui.qt.ui_base.BaseDialog`or in `traitsui.qt4.view_application.ViewApplication` after the event loop returns.
- in `pyface.ui.qt4.gui` somewhere
- in `pyface.ui.qt4.gui_application` somewhere
- in `pyface.ui.qt4.workbench` somewhere

plus potentially some other places (eg. possibly in Envisage).  Basically anywhere where we are starting/stopping the event loop.
